Add React 17 support

The current peerDependencies entry needs updated to include React 17. This is important because npm 7 now installs peerDepenencies automatically. So if you are using npm 7 and try to install Flux in a proejct using React 17, the following error occurs:

```
➜ npm i flux                            
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: flux-fix@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.0.2 || ^16.0.0-beta || ^16.0.0" from flux@3.1.3
npm ERR! node_modules/flux
npm ERR!   flux@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/coryhouse/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/coryhouse/.npm/_logs/2020-11-08T16_39_00_127Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -63,7 +63,7 @@
     "fbjs": "^3.0.0"
   },
   "peerDependencies": {
-    "react": "^15.0.2 || ^16.0.0-beta || ^16.0.0"
+    "react": "^15.0.2 || ^16.0.0-beta || ^16.0.0" || "^17.0.0"
   },
   "jest": {
     "modulePathIgnorePatterns": [

